src/App.tsx

```diff
@@ -66,41 +60,42 @@ export const App: React.FC = () => {
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length === 0 && (
+                {author && loading && !error && posts.length === 0 && (
                   <div className="notification is-warning" data-cy="NoPostsYet">
                     No posts yet
                   </div>
                 )}
 
-                {author && loaded && !hasError && posts.length > 0 && (
-                  <PostsList
-                    posts={posts}
-                    selectedPostId={selectedPost?.id}
-                    onPostSelected={setSelectedPost}
-                  />
+                {author && loading && !error && posts.length > 0 && (
+                  <PostsList />
                 )}
               </div>
             </div>
           </div>
 
-          <div
-            data-cy="Sidebar"
-            className={classNames(
-              'tile',
-              'is-parent',
-              'is-8-desktop',
-              'Sidebar',
-              {
-                'Sidebar--open': selectedPost,
-              },
-            )}
-          >
-            <div className="tile is-child box is-success ">
-              {selectedPost && <PostDetails post={selectedPost} />}
+          {author && selectedPost.selectedPost && (
+            <div
+              data-cy="Sidebar"
+              className={classNames(
+                'tile',
+                'is-parent',
+                'is-8-desktop',
+                'Sidebar',
+                {
+                  'Sidebar--open': selectedPost,
+                },
+              )}
+            >
+              <div className="tile is-child box is-success ">
+                {/* Pass selectedPost.selectedPost to PostDetails */}
+                <PostDetails post={selectedPost.selectedPost} />
```

Comment:
since you're using global state you don't need to pass this as a prop you can get it directly in component